File: CandyRainTolstonogova/CandyRainC++/CandyRain/Trash.cpp
#include "Trash.h"
#include <iostream>
#include <cstdlib>
#include <string>
#include "GameMain.h"
Trash::Trash(int x, int y, std::string spritepath, int spd, int num, bool ptouch, bool ftouch) :
    FallingObject(x, y, spritepath, spd, num, ptouch, ftouch) {
    std::cout << "Объект Trash создан" << std::endl;
}
Trash::~Trash() {
    std::cout << "Объект Trash уничтожен" << std::endl;
}
void Trash::PlayerCatchTrash(int playerX, int playerY) {
    if (abs(GetpositionX() - playerX) < GameMain::playersize && abs(GetpositionY() - playerY) < GameMain::playersize) {
        Setplayertouch(true);
        std::cout << "Trash пойман игроком!" << std::endl;
    }
}

File: CandyRainTolstonogova/CandyRainJava/Trash.java
public class Trash extends FallingObject {
    
    public Trash(int x, int y, String spritePath, int spd, int num, boolean pTouch, boolean fTouch) {
        super(x, y, spritePath, spd, num, pTouch, fTouch);
        System.out.println("Объект Trash создан");
    }
    
    public void playerCatchTrash(int playerX, int playerY) {
        if (Math.abs(getPositionX() - playerX) < GameMain.PLAYER_SIZE && 
            Math.abs(getPositionY() - playerY) < GameMain.PLAYER_SIZE) {
            setPlayerTouch(true);
            System.out.println("Trash пойман игроком!");
        }
    }
}